drivers/irqchip/irq-ts71xxweim.c

```diff
@@ -0,0 +1,167 @@
+// SPDX-License-Identifier: GPL-2.0
+
+#include <linux/types.h>
+#include <linux/module.h>
+#include <linux/irq.h>
+#include <linux/irqchip/chained_irq.h>
+#include <linux/irqdomain.h>
+#include <linux/of_device.h>
+
+#define TSWEIM_IRQ_STATUS	0x24
+#define TSWEIM_IRQ_MASK		0x48
+#define TSWEIM_NUM_FPGA_IRQ	32
+
+static struct tsweim_intc_priv {
+	void __iomem  *syscon;
+	struct irq_domain *irqdomain;
+	int irq;
+	u32 mask;
+} priv;
+
+static const struct of_device_id tsweim_intc_of_match_table[] = {
+	{.compatible = "technologic,ts71xxweim-intc", },
+	{},
+};
+MODULE_DEVICE_TABLE(of, tsweim_intc_of_match_table);
+
+static void ts7120_intc_mask(struct irq_data *d)
```

Comment:
What would happen if we just doubled-down on this being our WEIM-acccessible, irqchip for TS boards and get rid of the 7120 references?